### PR TITLE
Adjust dex to prow CI + add XSUAA connector

### PR DESCRIPTION
**Pull request by kubadz**
**Description**

Changes proposed in this pull request:
- adjust dex dockerfile and makefile
- add codeowners file
- add readme
- add changes required for xsuaa connector (cherry-picked from [piotrmsc/dex](https://github.com/piotrmsc/dex/tree/poc))

**Related issue(s)**

Related to [#kyma/6118](https://github.com/kyma-project/kyma/issues/6118)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
-COPY . /go/src/github.com/dexidp/dex
-RUN cd /go/src/github.com/dexidp/dex && make release-binary
+RUN apk add --update git gcc libc-dev
 
-FROM alpine:3.9
+WORKDIR ${BASE_DIR}
+
+COPY . ${BASE_DIR}/
+
+RUN GO111MODULE=on go mod tidy -v
```
Comment (Demonsthere) [inline Dockerfile]: Maybe move the env to an `ENV` field

File: README.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+> This repository is a fork. All custom changes are done on `kyma-master` branch.
+>
+> [This document](CONTRIBUTING.md) describes the process of contributing to this repository.
```
Comment (jakkab) [inline README.md]: dead link

Comment (kubadz) [reply README.md]: It is intentionally dead. I will create another PR with the contribution guide.

**Pull request merged**